src/game_inventory.cpp

```diff
@@ -92,62 +118,222 @@ int game::inv_for_unequipped( const std::string &title )
     }, _( "You don't have any items to wear." ) );
 }
 
-item_location game::inv_map_splice( item_filter filter, const std::string &title, int radius,
+item_location game::inv_map_splice( const item_filter &filter, const std::string &title, int radius,
                                     const std::string &none_message )
 {
-    return inv_map_splice( convert_filter( filter ), title, radius, none_message );
+    return inv_map_splice( [ &filter ]( const item_location & location ) {
+        return filter( *location.get_item() );
+    }, title, radius, none_message );
 }
 
-item_location game::inv_map_splice( item_location_filter filter, const std::string &title,
+item_location game::inv_map_splice( const item_location_filter &filter, const std::string &title,
                                     int radius,
                                     const std::string &none_message )
 {
-    u.inv.restack( &u );
-    u.inv.sort();
+    return inv_internal( u, inventory_filter_preset( filter ),
+                         title, radius, none_message );
+}
 
-    inventory_pick_selector inv_s( u, title, filter );
+class liquid_inventory_preset: public inventory_selector_preset
+{
+    public:
+        liquid_inventory_preset( const item &liquid ) : liquid( liquid ) {
```

Comment:
Always prefer `itype_id` over `item` where possible. In many cases the caller is concerned about a certain liquid not a specific instance of it.